dts/bindings/usb/usb.yaml

```diff
@@ -0,0 +1,50 @@
+---
+title: USB Base Structure
+id: USB
+version: 0.1
+
+description: >
+    This binding gives the base structures for all USB devices
+
+properties:
+  - compatible:
+      type: string
+      category: required
+      description: compatible strings
+
+  - clock-frequency :
+      type: int
+      category: optional
+      description: Initial clock frequency in Hz
+      generation: define
+
+  - clocks:
+      type: array
+      category: required
+      description: Clock gate information
+      generation: define
+
+  - num-bidir-endpoints:
+      type: int
+      category: required
```

Comment:
Do we want to make these required.  Wonder about the case that the value is 0.  Seems like we would just not have the property than.

Reply:
Or just set the value to 0. For example, the OTG FS for F4 has 1 bi-directional, 3 IN, 3 OUT and USB FS for F1 has 8 bi-directional. The stm32 usb driver needs the total number of in/out endpoints (including bi-directional). Now it is done with 3 macros for both cases. If the properties are optional the drivers needs ifdefs to handle all the cases. It will make the code harder to read.